open-ce/test_feedstock.py

```diff
@@ -0,0 +1,223 @@
+US Government Users Restricted Rights - Use, duplication or
+disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
+*****************************************************************
+"""
+
+import datetime
+import os
+import tempfile
+import sys
+import subprocess
+
+import yaml
+
+import utils
+from utils import OpenCEError
+
+class TestCommand():
+    """
+    Contains a test to run within a given conda environment.
+
+    Args:
+        name (str): The name describing the test.
+        conda_env (str): The name of the conda environment that the test will be run in.
+        bash_command (str): The bash command to run.
+        create_env (bool): Whether this is the command to create a new conda environment.
+        clean_env (bool): Whether this is the command to remove a conda environment.
+    """
+    #pylint: disable=too-many-arguments
+    def __init__(self, name, conda_env=None, bash_command="", create_env=False, clean_env=False):
+        self.bash_command = bash_command
+        self.conda_env = conda_env
+        self.name = name
+        self.create_env = create_env
+        self.clean_env = clean_env
+
+    def get_test_command(self, conda_env_file=None):
+        """"
+        Returns a string of the test command.
+
+        Args:
+            conda_env_file (str): The name of the original conda environment file.
+                                  This is only needed when create_env is True.
+        """
+        output = ""
+        if self.create_env:
+            output += "conda env create -f " + conda_env_file + " -n " + self.conda_env + "\n"
+            return output
+
+        if self.clean_env:
+            output += "conda env remove -y -n " + self.conda_env + "\n"
+            return output
+
+        output += "CONDA_BIN=$(dirname $(which conda))\n"
+        output += "source ${CONDA_BIN}/../etc/profile.d/conda.sh\n"
+        output += "conda activate " + self.conda_env + "\n"
+        output += self.bash_command + "\n"
+
+        return output
+
+    def run(self, conda_env_file):
+        """
+        Runs the test.
+
+        Creates a temporary bash file, and writes the contents to `get_test_command` into it.
+        Runs the generated bash file.
+        Removes the temporary bash file.
+
+        Args:
+            conda_env_file (str): The name of the original conda environment file.
+        """
+        print("Running: " + self.name)
+        # Create file containing bash commands
+        with tempfile.NamedTemporaryFile(mode='w+t', dir=os.getcwd(), delete=False) as temp:
+            temp.write("set -e\n")
+            temp.write(self.get_test_command(conda_env_file))
+            temp_file_name = temp.name
+
+        # Execute file
+        test_process = subprocess.run(["bash", temp_file_name], stdout=subprocess.PIPE, check=False,
+                                      stderr=subprocess.STDOUT, universal_newlines=True)
+
+        # Remove file containing bash commands
```

Comment:
Maybe there are scenarios where we'd want this file to be written to disk? Perhaps if a test failed?

Reply:
That might be the case, but that could end up leaving kind of a mess. I've opened an issue to choose a working directory for running tests (#139). Once that is completed, it might make sense to add an option to keep files around.

Reply:
Ok. I added a comment to #139 for this.